pubsub/topics/main.go

```diff
@@ -145,6 +154,43 @@ func publish(client *pubsub.Client, topic, msg string) error {
 	return nil
 }
 
+func publishThatScales(client *pubsub.Client, topic string, n uint64) error {
+	ctx := context.Background()
+	var wg sync.WaitGroup
+	var n_err uint64
+	// [START pubsub_publish_with_error_handling_that_scales]
+	t := client.Topic(topic)
+
+	for i := 0; i < int(n); i++ {
+		result := t.Publish(ctx, &pubsub.Message{
+			Data: []byte("Message No. " + strconv.Itoa(i)),
+		})
+
+		wg.Add(1)
+		go func(i int) {
+			defer wg.Done()
+			// The Get method blocks until the result is returned and a
+			// server-generated ID is returned for the published message.
+			id, err := result.Get(ctx)
```

Comment:
I think you also want to pass the `result` as a function argument to give it a copy of the result object, otherwise all the goroutines will be reading the latest value of `result` rather than their own copy.

Reply:
Thanks David for pointing that out!